main.py

```diff
@@ -1,16 +1,34 @@
-# This is a sample Python script.
+from flask import Flask, request, render_template, redirect, url_for, make_response
 
-# Press Shift+F10 to execute it or replace it with your code.
-# Press Double Shift to search everywhere for classes, files, tool windows, actions, and settings.
+app = Flask(__name__)
 
+@app.route('/')
+def index():
+    return render_template("start.html")
 
-def print_hi(name):
-    # Use a breakpoint in the code line below to debug your script.
-    print(f'Hi, {name}')  # Press Ctrl+F8 to toggle the breakpoint.
+@app.post('/start/')
+def start():
+    response = make_response(render_template("start.html"))
+    name = request.form.get('name')
+    mail = request.form.get('mail')
+    response = make_response(redirect(url_for('hello', name=name)))
+    response.set_cookie('name', name)
+    response.set_cookie('mail', mail)
+    return response
 
 
-# Press the green button in the gutter to run the script.
-if __name__ == '__main__':
-    print_hi('PyCharm')
+@app.route('/hello/<name>')
+def hello(name):
+    return render_template('hello.html', name=name)
 
-# See PyCharm help at https://www.jetbrains.com/help/pycharm/
+
+@app.post('/exit/')
+def exit():
```

Comment:
Тут также
```suggestion
@app.post('/hello/<name>')
def exit(name):
```

Но тогда появляется неиспользуемый параметр `name`. Чтобы такого не  было все функции можно объединить попарно в две, проверяя в каждой тип запроса